internal/core/command/controller/messaging/external/external.go

```diff
@@ -0,0 +1,139 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package external
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+
+	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/command/application"
+	"github.com/edgexfoundry/edgex-go/internal/core/command/container"
+)
+
+const (
+	RequestQueryTopic  = "RequestQueryTopic"
+	ResponseQueryTopic = "ResponseQueryTopic"
+)
+
+func OnConnectHandler(dic *di.Container) mqtt.OnConnectHandler {
+	return func(client mqtt.Client) {
+		lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+		config := container.ConfigurationFrom(dic.Get)
+		externalTopics := config.MessageQueue.External.Topics
+		qos := config.MessageQueue.External.QoS
+		retain := config.MessageQueue.External.Retain
+
+		requestQueryTopic := externalTopics[RequestQueryTopic]
+		responseQueryTopic := externalTopics[ResponseQueryTopic]
+		if token := client.Subscribe(requestQueryTopic, qos, commandQueryHandler(responseQueryTopic, qos, retain, dic)); token.Wait() && token.Error() != nil {
+			lc.Errorf("could not subscribe to topic '%s': %s", responseQueryTopic, token.Error().Error())
+
+			return
+		}
+	}
+}
+
+func commandQueryHandler(responseTopic string, qos byte, retain bool, dic *di.Container) mqtt.MessageHandler {
+	return func(client mqtt.Client, message mqtt.Message) {
+		var errorMessage string
+		var responseEnvelope types.MessageEnvelope
+		lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+
+		requestEnvelope, err := types.NewMessageEnvelopeFromJSON(message.Payload())
+		if err != nil {
+			responseEnvelope = types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, err.Error())
+			publishMessage(client, responseTopic, qos, retain, responseEnvelope, lc)
+			return
+		}
+
+		// example topic scheme: edgex/commandquery/request/#
```

Comment:
Actually the received topic doesn't have the wild card `#`, 
```suggestion
		// example topic scheme: edgex/commandquery/request/<device>
```